gridstatus/spp.py

```diff
@@ -193,6 +205,169 @@ def get_load_forecast(self, date, forecast_type="MID_TERM", verbose=False):
 
         return current_day_forecast
 
+    def get_solar_and_wind_forecast_short_term(self, date, end=None, verbose=False):
+        """
+        Returns solar and wind generation forecast for +4 hours in 5 minute intervals.
+        Include actuals for past day in 5 minute intervals.
+
+        Arguments:
+            date (pd.Timestamp|str): date to get data for. Supports "latest" and "today"
+            verbose (bool): print info
+
+        Returns:
+            pd.DataFrame: forecast as dataframe.
+        """
+        # The short_term forecast is delayed up to 2 minutes.
+        buffer_minutes = 2
+
+        now = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        if date in ["latest", "today"]:
+            date = now
+
+        date = utils._handle_date(date, self.default_timezone)
+        date = min(now, date)
+
+        if end:
+            end = utils._handle_date(end, self.default_timezone)
+            # Make sure we don't request files that don't exist yet
+            end = min(now, end)
+
+        df = self._retrieve_solar_and_wind_forecast_short_term(
+            date,
+            end,
+            verbose,
+        )
+
+        df["Forecast Type"] = "SHORT_TERM"
+
+        if end:
+            return df[df["Publish Time"].between(date, end)]
+
+        return df
+
+    @support_date_range("5_MIN")
+    def _retrieve_solar_and_wind_forecast_short_term(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """Solar and wind forecast for +4 hours by 5 minute interval."""
+        url = self._short_term_solar_and_wind_url(date.floor("5T"))
+
+        df = pd.read_csv(url)
+
+        # According to the docs, the end time col should be GMTIntervalEnd, but it's
+        # only GMTInterval in the data
+        df = self._post_process_solar_and_wind_forecast(
+            df,
+            url,
+            end_time_col="GMTInterval",
+            interval_duration=pd.Timedelta(minutes=5),
+        )
+
+        return df
+
+    def get_solar_and_wind_forecast_mid_term(self, date, end=None, verbose=False):
+        """
+        Returns solar and wind generation forecast for +7 days in hourly intervals.
+
+        Arguments:
+            date (pd.Timestamp|str): date to get data for. Supports "latest" and "today"
+            verbose (bool): print info
+
+        Returns:
+            pd.DataFrame: forecast as dataframe.
+        """
+        # The MID_TERM forecast is delayed up to 10 minutes.
+        buffer_minutes = 10
+
+        now = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        if date in ["latest", "today"]:
```

Comment:
i think the handle of "today" should be to get all forecasts published today. i think if you used `@support_date_range` to decorate this function, it would handle that automatically by calling this function for each hour of the day

Reply:
also, we can error on "latest" if there isn't a way to directly go to a latest file

Reply:
> there isn't a way to directly go to a latest file

It seems relatively simple to set the date to the current timestamp to get the latest file.

Reply:
i just think of it as one more like of code we need to maintain. if user want this, it's easy enough for them to call "pd.Timestamp.now()" in their own code and pass that as the parameter. if there was a specific latest file like other endpoints, i think it makes more sense for us to implement handling